hack/gatewayapi-conformance.sh

```diff
@@ -32,22 +32,30 @@ cd "${CLONE_DIR}"
 # find the branch of gateway-api repo
 RELEASE_VERSION=$(\grep -oP "^\d+\.\d+" <<<"${BUNDLE_VERSION#v}")
 BRANCH="release-${RELEASE_VERSION}"
-echo "gateway-api repo branch: ${BRANCH}"
+echo "gateway-api repo branch \"${BRANCH}\" into ${CLONE_DIR}..."
 
 # clone the branch that matched the installed CRDs version
 # branch release-1.0 -> bundle-version v1.0.0
 # branch release-1.1 -> bundle-version v1.1.1
 # branch release-1.2 -> bundle-version v1.2.1
 git clone --branch "${BRANCH}" https://github.com/kubernetes-sigs/gateway-api
 cd gateway-api
+
+if [[ "$BUNDLE_VERSION" = "v1.2.1" ]]; then
+    echo "Cherry-picking fix for CoreDNS deployment issue in Gateway API v1.2.1"
+    git fetch origin f64c54a3606c8eee5b4c85b1c5f8f0d3cf3470ca
+    git cherry-pick f64c54a3606c8eee5b4c85b1c5f8f0d3cf3470ca
+fi
+
 echo "Go version: $(go version)"
 go mod vendor
 
-# modify default timeout of "MaxTimeToConsistency" to make tests pass on AWS
-# because the AWS ELB needs extra ~60s for DNS propagation
-sed -i "s/MaxTimeToConsistency:              30/MaxTimeToConsistency:              90/g" conformance/utils/config/timeout.go
+# Modify the default timeout of "MaxTimeToConsistency" to make tests pass on AWS
+# because the AWS ELB needs an extra ~60s for DNS propagation.  See
+# <https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/best-practices-dns.html#DNS_change_propagation>.
+sed -i -e '/MaxTimeToConsistency:/ s/30/90/' conformance/utils/config/timeout.go
 
-SUPPORTED_FEATURES="Gateway,HTTPRoute,ReferenceGrant,GatewayPort8080,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,HTTPRouteResponseHeaderModification,HTTPRoutePortRedirect,HTTPRouteSchemeRedirect,HTTPRoutePathRedirect,HTTPRouteHostRewrite,HTTPRoutePathRewrite,HTTPRouteRequestMirror,HTTPRouteRequestMultipleMirrors,HTTPRouteBackendProtocolH2C,HTTPRouteBackendProtocolWebSocket"
+SUPPORTED_FEATURES='Gateway,HTTPRoute,GatewayPort8080,GatewayStaticAddresses,GatewayHTTPListenerIsolation,HTTPRouteBackendRequestHeaderModification,HTTPRouteQueryParamMatching,HTTPRouteMethodMatching,HTTPRouteResponseHeaderModification,HTTPRoutePortRedirect,HTTPRouteSchemeRedirect,HTTPRoutePathRedirect,HTTPRouteHostRewrite,HTTPRoutePathRewrite,HTTPRouteRequestMirror,HTTPRouteRequestMultipleMirrors,HTTPRouteRequestTimeout,HTTPRouteBackendTimeout,HTTPRouteParentRefPort,HTTPRouteBackendProtocolH2C,HTTPRouteBackendProtocolWebSocket,GRPCRoute,ReferenceGrant'
```

Comment:
Seems the `HTTPRouteDestinationPortMatching` is missing, and maybe remove `GatewayStaticAddresses` since it is hard to test here.

Reply:
`HTTPRouteDestinationPortMatching` is experimental: https://github.com/kubernetes-sigs/gateway-api/blob/v1.2.1/pkg/features/httproute.go#L100-L104 

I'll remove `GatewayStaticAddresses` for the time being.